tests/Tests/500__jupyterhub/custom-image.robot

```diff
@@ -1,51 +1,135 @@
+Suite Teardown   End Web Test
 Force Tags       JupyterHub
 
 
+
 *** Variables ***
 ${YAML} =         tests/Resources/Files/custom_image.yaml
-${IMG_NAME} =     custom-test-image
+${IMG_NAME} =            custom-test-image
+${IMG_URL} =             quay.io/thoth-station/s2i-lab-elyra:v0.1.1
+${IMG_DESCRIPTION} =     Testing Only This image is only for illustration purposes, and comes with no support. Do not use.
+&{IMG_SOFTWARE} =        Software1=x.y.z
+&{IMG_PACKAGES} =        elyra=2.2.4    foo-pkg=a.b.c
+# Place holder for the imagestream name of BYON notebook created for this test run
+${IMAGESTREAM_NAME}=
 
 
 *** Test Cases ***
+Verify Admin User Can Access Custom Notebook Settings
+    Set Library Search Order  SeleniumLibrary
+    Launch Dashboard    ocp_user_name=${TEST_USER.USERNAME}    ocp_user_pw=${TEST_USER.PASSWORD}
+    ...    ocp_user_auth_type=${TEST_USER.AUTH_TYPE}    dashboard_url=${ODH_DASHBOARD_URL}
+    ...    browser=${BROWSER.NAME}    browser_options=${BROWSER.OPTIONS}
+    Open Notebook Images Page
+
 Verify Custom Image Can Be Added
-    [Documentation]    Applies the YAML and Gets the ImageStream
+    [Documentation]    Imports the custom image via UI
     ...                Then loads the spawner and tries using the custom img
     [Tags]    Tier2
     ...       ODS-1208
     Apply Custom ImageStream And Check Status
     Get ImageStream Metadata And Check Name
-    Begin Web Test
+    Verify Image Name  ${IMG_NAME}
+    Verify Image Description  ${IMG_NAME}  ${IMG_DESCRIPTION}
+    Verify User  ${IMG_NAME}  ${TEST_USER.USERNAME}
+    Launch JupyterHub Spawner From Dashboard
+
+    # These keywords need to be reworked to function here
+    #${spawner_description}=  Fetch Image Tooltip Description  ${IMAGESTREAM_NAME}
+    #${spawner_packages}=  Fetch Image Tooltip Info  ${IMAGESTREAM_NAME}
+    #${spawner_software}=  Fetch Image Description Info  ${IMAGESTREAM_NAME}
+    #Should Match  ${spawner_description}  ${IMG_DESCRIPTION}
+    #Should Match  ${spawner_software}  ${IMG_SOFTWARE}
+    #Should Match  ${spawner_packages}  ${IMG_PACKAGES}
+
+    Spawn Notebook With Arguments  image=${IMAGESTREAM_NAME}  size=Default
+    [Teardown]  Custom Image Teardown
+
+Test Duplicate Image
+    [Documentation]  Test adding two images with the same name (should fail)
+    Apply Custom ImageStream And Check Status
+    Sleep  1
+    Import New Image    ${IMG_URL}    ${IMG_NAME}    ${IMG_DESCRIPTION}
+    ...    software=${IMG_SOFTWARE}
+    ...    packages=${IMG_PACKAGES}
+    RHODS Notification Drawer Should Contain  Unable to add notebook image ${IMG_NAME}
+    Sleep  1
+    Delete Image  ${IMG_NAME}
+    Reset Image Name
+
+Test Bad Image URL
+    [Documentation]  Test adding an image with a bad repo URL (should fail)
+    ${OG_URL}=  Set Variable  ${IMG_URL}
+    ${IMG_URL}=  Set Variable  quay.io/RandomName/RandomImage:v1.2.3
+    Set Global Variable  ${IMG_URL}  ${IMG_URL}
+    Apply Custom ImageStream And Check Status
```

Comment:
you would spare some lines of code if you add args to this kw

Reply:
Not sure what you meant here

Reply:
sorry it was a very vague comment xD 
In a kw later you have
``` 
Apply Custom ImageStream And Check Status
    Sleep  1
    Import New Image    ${IMG_URL}    ${IMG_NAME}    ${IMG_DESCRIPTION}
    ...    software=${IMG_SOFTWARE}
    ...    packages=${IMG_PACKAGES}
   ```
If `Apply Custom ImageStream And Check Status` would have arguments, you would just call the kw twice here instead of using `Import New Image` again. 

(actually you wouldn't spare lines of code at the end of the day).
It's is more a aesthetic adjustment maybe. Not a big deal, feel free to discard this

Reply:
I think you mean in `Test Duplicate Image`, the keyword above this one.
The way it's currently written I know it's a slight repetition, but it was just the easiest way to deal with the global variable and still achieve a duplicate img.

Reply:
yes I refer to that `Test Duplicate Image`.

it's okay anyway